packages/suite/src/components/suite/Preloader/index.tsx

```diff
@@ -35,50 +33,21 @@ import {
     Welcome,
 } from '@suite-views';
 
-const mapStateToProps = (state: AppState) => ({
-    loading: state.suite.loading,
-    loaded: state.suite.loaded,
-    error: state.suite.error,
-    device: state.suite.device,
-    transport: state.suite.transport,
-    router: state.router,
-    actionModalContext: state.modal.context,
-    // TODO: THIS IS SO WRONG
-    // Without this the app will get stuck on "checking balances" in case of entering diff passphrase in passphrase confirmation
-    // So even we do not use discovery anywhere, this component really depends on it
-    // eslint-disable-next-line react/no-unused-prop-types
-    discovery: state.wallet.discovery,
-});
-
-const init = () =>
-    ({
-        type: SUITE.INIT,
-    } as const);
-
-const mapDispatchToProps = (dispatch: Dispatch) =>
-    bindActionCreators(
-        {
-            init,
-            getDiscoveryAuthConfirmationStatus: discoveryActions.getDiscoveryAuthConfirmationStatus,
-            goto: routerActions.goto,
-            closeModalApp: routerActions.closeModalApp,
-            getBackgroundRoute: routerActions.getBackgroundRoute,
-        },
-        dispatch,
-    );
-
-type Props = ReturnType<typeof mapStateToProps> &
-    ReturnType<typeof mapDispatchToProps> & {
-        children: React.ReactNode;
-    };
-
-type SuiteAppStateProps = Pick<
-    Props,
-    'loaded' | 'transport' | 'device' | 'router' | 'getDiscoveryAuthConfirmationStatus'
->;
-const getSuiteApplicationState = (props: SuiteAppStateProps) => {
-    const { loaded, transport, device, getDiscoveryAuthConfirmationStatus, router } = props;
+type SuiteAppStateProps = {
+    loaded: boolean;
+    transport: AppState['suite']['transport'];
+    device: AppState['suite']['device'];
+    router: AppState['router'];
+    getDiscoveryStatus: () => any;
```

Comment:
why not type it properly? `() => DiscoveryStatus | undefined` (exported from `@wallet-types`);
)

Reply:
good catch, i was in a hurry and forgot about it :)

Reply:
done